server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/VacationRoutes.java

```diff
@@ -0,0 +1,199 @@
+                          UsersRepository usersRepository, JsonTransformer jsonTransformer) {
+        this.vacationRepository = vacationRepository;
+        this.notificationRegistry = notificationRegistry;
+        this.usersRepository = usersRepository;
+        this.jsonTransformer = jsonTransformer;
+        this.jsonExtractor = new JsonExtractor<>(VacationDTO.class, new JavaTimeModule());
+    }
+
+    @Override
+    public String getBasePath() {
+        return VACATION;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.get(VACATION + SEPARATOR + USER_NAME, this::getVacation, jsonTransformer);
+        service.post(VACATION + SEPARATOR + USER_NAME, this::updateVacation);
+        service.patch(VACATION + SEPARATOR + USER_NAME, this::setVacation);
+        service.delete(VACATION + SEPARATOR + USER_NAME, this::deleteVacation);
+    }
+
+    public VacationDTO getVacation(Request request, Response response) {
+        testUserExists(request);
+        try {
+            AccountId accountId = AccountId.fromString(request.params(USER_NAME));
+            Vacation vacation = vacationRepository.retrieveVacation(accountId).block();
+            return VacationDTO.from(vacation);
+        } catch (IllegalStateException e) {
+            LOGGER.info("Invalid get on user vacation", e);
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Invalid get on user vacation")
+                .cause(e)
+                .haltError();
+        }
+    }
+
+    public String updateVacation(Request request, Response response) {
+        testUserExists(request);
+        try {
+            AccountId accountId = AccountId.fromString(request.params(USER_NAME));
+            VacationDTO vacationDto = jsonExtractor.parse(request.body());
+            VacationPatch vacationPatch = VacationPatch.builder()
+                .subject(updateTo(vacationDto.getSubject()))
+                .textBody(updateTo(vacationDto.getTextBody()))
+                .htmlBody(updateTo(vacationDto.getHtmlBody()))
+                .fromDate(updateTo(vacationDto.getFromDate()))
+                .toDate(updateTo(vacationDto.getToDate()))
+                .isEnabled(updateTo(vacationDto.getEnabled()))
+                .build();
+            vacationRepository.modifyVacation(accountId, vacationPatch)
+                .then(notificationRegistry.flush(accountId)).block();
+            return Responses.returnNoContent(response);
+        } catch (JsonExtractException e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Malformed JSON input")
+                .cause(e)
+                .haltError();
+        }
```

Comment:
All the handling of `JsonExtractException` should already be done by the default error handler of the WebAdminServer. We likely can remove such statements here.

Reply:
Just checked, it actually does return 400 on bad JSON. I can remove the catch clause and just declare the method `throws JsonExtractException`. Good to know :-)